### DISCUSSION TITLE
.hex file

### DISCUSSION THREAD

**Discussion by southwcm** [fra589/grbl-Mega-5X#236] [Q&A] [answered]
I can't seem to find the .hex file to run XLoader. Any advice?

**Comment by fra589** [comment, marked as answer]
Hi @southwcm,  
  
Considering that there can be many many different configurations (number of axes, naming and cloning of axes, change of pinouts, ...) it does not seem possible to me to provide a hex file for all.  
You must configure and compile yourself the version corresponding to your needs.  
The easiest for most users is to use the [Arduino software](https://www.arduino.cc/en/software). The procedure to build Grbl-Mega-5X and upload it to the card is described in the wiki here:  
https://github.com/fra589/grbl-Mega-5X/wiki/Compiling-grbl-Mega-5X  
  
@++;  
Gauthier.

**Reply by southwcm** [reply]
Okay, I'm trying to follow the instructions you provided with what I think is the latest Grbl-Mega. I'm trying to load this onto my Arduino Mega 2560 to power a Ramps 1.4 shield set.  Is the latest version grbl-Mega-5X-edge? When I try to add the library, it says "Specific folder/zip file does not contain a valid library." I made sure to delete the previous GRBL libraries as recommended. Your help is greatly appreciated as I am relatively new to the Github/Arduino world.

**Reply by fra589** [reply]
Hello @southwcm,  
  
Are you under Windows or under Linux?  
  
In the downloaded zip file, there is a directory named "grbl" under the global "grbl-Mega-5X-edge" directory. You need to copy only the "grbl" sub-directory to the libray location of your Arduino's system.  
  
@++;  
Gauthier.  
![image](https://user-images.githubusercontent.com/19529154/149615055-ea2320f4-d6bd-4829-8fe9-91f75016e116.png)

**Reply by fra589** [reply]
Yes, the grbl-Mega-5X version you downloaded on https://github.com/fra589/grbl-Mega-5X/ is the latest one.

**Comment by Salehdanaee** [comment]
Hello
No matter what I do, I can't compile the program...

The grbl-XXX file is not in my code folder...


When the zip file in this way
Click the Sketch drop-down menu, navigate to Include Library and select Add .ZIP Library. The Add .ZIP Library command supports both a .ZIP file or a folder. In our case, there is no .ZIP file.

 When I install it, I get the following error

Error: 13 INTERNAL: Library install failed: moving extracted archive to destination dir: library not valid

**Reply by RaphaelDives** [reply]
Hey,
the answer is just above…

> In the downloaded zip file, there is a directory named "grbl" under the global "grbl-Mega-5X-edge" directory. You need to copy only the "grbl" sub-directory to the libray location of your Arduino's system.

In case you don’t know where your Arduino library is located. Google is your friend… 
I’m my case (win10) it is in my user folder/Documents/Arduino/libraries/
Cheers Raphael

**Comment by fra589** [comment]
Read all details of this:
https://github.com/fra589/grbl-Mega-5X/wiki/Compiling-grbl-Mega-5X

Sorry, it's not a newbies kids youtube video...

@++;
Gauthier.
